File: server/src/models/Booking.js
import { DataTypes } from 'sequelize';
import sequelize from '../config/database.js';

const Booking = sequelize.define('Booking', {
  id: {
    type: DataTypes.UUID,
    defaultValue: DataTypes.UUIDV4,
    primaryKey: true
  },
  booking_number: {
    type: DataTypes.STRING,
    unique: true,
    allowNull: false
  },
  user_id: {
    type: DataTypes.UUID,
    allowNull: true
  },
  customer_name: {
    type: DataTypes.STRING,
    allowNull: false
  },
  customer_phone: {
    type: DataTypes.STRING,
    allowNull: false
  },
  customer_email: {
    type: DataTypes.STRING,
    allowNull: true
  },
  booking_date: {
    type: DataTypes.DATEONLY,
    allowNull: false
  },
  booking_time: {
    type: DataTypes.TIME,
    allowNull: false
  },
  guests_count: {
    type: DataTypes.INTEGER,
    allowNull: false
  },
  table_preference: {
    type: DataTypes.STRING,
    allowNull: true
  },
  special_requests: {
    type: DataTypes.TEXT,
    allowNull: true
  },
  status: {
    type: DataTypes.ENUM(
      'pending',      // ожидает подтверждения
      'confirmed',    // подтвержден
      'seated',       // гости за столом
      'completed',    // завершен
      'cancelled',    // отменен
      'no_show'       // не пришли
    ),
    defaultValue: 'pending'
  },
  confirmed_by: {
    type: DataTypes.STRING, // имя менеджера (Анна)
    allowNull: true
  },
  confirmation_notes: {
    type: DataTypes.TEXT,
    allowNull: true
  }
}, {
  tableName: 'bookings',
  timestamps: true
});

export default Booking;

File: flutter/cafeteria_app/lib/models/booking.dart
class BookingRequest {
  final String bookingDate;
  final String bookingTime;
  final int guestsCount;
  final String tablePreference;
  final String specialRequests;
  final String customerName;
  final String customerPhone;
  final String customerEmail;

  BookingRequest({
    required this.bookingDate,
    required this.bookingTime,
    required this.guestsCount,
    required this.tablePreference,
    required this.specialRequests,
    required this.customerName,
    required this.customerPhone,
    required this.customerEmail,
  });

  Map<String, dynamic> toJson() => {
        'booking_date': bookingDate,
        'booking_time': bookingTime,
        'guests_count': guestsCount,
        'table_preference': tablePreference,
        'special_requests': specialRequests,
        'customer_name': customerName,
        'customer_phone': customerPhone,
        'customer_email': customerEmail,
      };
}

class BookingResponse {
  final int id;
  final String bookingNumber;
  final String status;

  BookingResponse({
    required this.id,
    required this.bookingNumber,
    required this.status,
  });

  factory BookingResponse.fromJson(Map<String, dynamic> json) {
    return BookingResponse(
      id: json['id'] as int,
      bookingNumber: json['booking_number'] as String,
      status: json['status'] as String? ?? 'pending',
    );
  }
}
